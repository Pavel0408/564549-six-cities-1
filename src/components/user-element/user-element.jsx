import React from "react";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import {userPropTypes} from "../../prop-types/user-prop-types";

export const UserElement = (props) => {
  const {onFetchFavorite, user} = props;
  return <Link
    className="header__nav-link header__nav-link--profile"
    to={`/favorites`} onClick={onFetchFavorite}> <div className="header__avatar-wrapper user__avatar-wrapper">
      <img src={`https://es31-server.appspot.com/six-cities${user.avatar}`}/>
      <span className="header__user-name user__name">{user.email}</span>
    </div>
  </Link>;
};

UserElement.propTypes = {
  onFetchFavorite: PropTypes.func,
  user: userPropTypes
};
