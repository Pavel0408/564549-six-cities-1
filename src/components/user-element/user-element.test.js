import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

import {UserElement} from "./user-element";

const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  avatar: `img/1.png`,
  isPro: false
};

describe(`testing UserElement render`, () => {
  it(`UserElement is render correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <UserElement
            user={user}
          />
        </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


