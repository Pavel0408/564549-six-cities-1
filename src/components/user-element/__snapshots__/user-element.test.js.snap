// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserElement render UserElement is render correctly 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/favorites"
  onClick={[Function]}
>
   
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  >
    <img
      src="https://es31-server.appspot.com/six-citiesimg/1.png"
    />
    <span
      className="header__user-name user__name"
    >
      Oliver.conner@gmail.com
    </span>
  </div>
</a>
`;
